src/app/loadout-builder/PerkPicker.tsx

```diff
@@ -124,6 +133,18 @@ function mapStateToProps() {
     }
   );
 
+  function getAlreadyInsertedIndex(plugSetItem, defs) {
```

Comment:
```suggestion
  function getAlreadyInsertedIndex(plugSetItem:DestinyItemPlug, defs:D2ManifestDefinitions) {
```
properly typing the arguments will ensure that inside the function,
typescript understands things like the fact that `rule.failureMessage` will be a `string`

telling TS that `alreadyInsertedRule` is a `DestinyPlugRuleDefinition` doesn't serve a purpose unless TS knows what type is on the other side of that `===`

you could also just skip typing for `alreadyInsertedRule` because you're really just using it as a string holder right now.

`rule.failureMessage(TS knows this is a string) === alreadyInsertedFailureMessage(TS knows this is a string)`
is just as typesafe

Reply:
Typing `alreadyInsertedRule` was actually a hang over from something else I tried, but you're right. I've been trying to be explicit with my typing but the old js habits kick in a lot. In either case my new proposed change deletes this.